common/cosmwasm-smart-contracts/service-provider-directory/src/msg.rs

```diff
@@ -0,0 +1,138 @@
+use crate::{NymAddress, Service, ServiceId, ServiceType};
+use cosmwasm_std::{Addr, Coin};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub struct InstantiateMsg {
+    pub deposit_required: Coin,
+}
+
+impl InstantiateMsg {
+    pub fn new(deposit_required: Coin) -> Self {
+        Self { deposit_required }
+    }
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    Announce {
+        nym_address: NymAddress,
+        service_type: ServiceType,
+    },
+    Delete {
+        service_id: ServiceId,
```

Comment:
imo there should be also an option to delete by `NymAddress` since if you announce your service provider, you're not going to even know the `ServiceId` unless you query for it

Reply:
Good point